public/app/plugins/datasource/loki/components/LokiContextUi.tsx

```diff
@@ -163,6 +181,31 @@ export function LokiContextUi(props: LokiContextUiProps) {
 
   return (
     <div className={styles.wrapper}>
+      <Button
+        title="Revert to initial log context query."
+        icon="history-alt"
+        variant="secondary"
+        className={cx(
+          styles.iconButton,
+          css`
+            visibility: ${isInitialQuery ? 'hidden' : 'visible'};
+          `
+        )}
+        tooltip="Revert to initial log context query."
+        onClick={(e) => {
+          e.stopPropagation();
+          setContextFilters((contextFilters) => {
+            return contextFilters.map((contextFilter) => {
+              return {
+                ...contextFilter,
+                // For revert to initial query we need to enable all labels and disable all parsed labels
+                enabled: !contextFilter.fromParser,
+              };
+            });
+          });
```

Comment:
I don't think it is correct because you want to `setContextFilters` based on previous `contextFilters`, therefore we need to use function instead of just a new array of contextFilters. But I was able to remove couple of lines in here https://github.com/grafana/grafana/pull/68484/commits/43ab650f9e2884321806afc23e7e8cee09d59a91

Reply:
You already have the previous `contextFilters` in the state variable https://github.com/grafana/grafana/pull/68484/files/243c75187ba6c8851629b6c0c41edfeaf98e7e51#diff-482a82afb5742195c02896eecc365131b6283b498d80446168cb7daf64286f3cR85 , calling `setContextFilters` with a callback is not mandatory for that.
In any case, this also works.